### PR TITLE
Updating existing user documentation with more details

### PR DESCRIPTION
**Pull request by missymessa**
This PR expands upon the existing documentation as we've rolled out more features since it was originally created. Feedback I'm looking for from reviewers: 

- If you've been involved with this epic, please check for accuracy in the statements made. Point out anything you feel is missing that I should add or expand upon. 
- If you haven't been involved with this epic, please check that it makes sense from a customer's point-of-view (for dnceng folks, it's currently enabled on Arcade and Arcade Services, so we technically are customers, too!). 
- If you think of other FAQ items to add, tell me about them and I'll address it. 

Thanks everyone!

### CODE REVIEW

**Comment by missymessa** [comment]
Need to fix that image with the weird spacing.

**Review by ChadNedzlek — APPROVED** [review]

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Build Analysis
 
-The Build Analysis Check is a service to improve the Pull Request experience by highlighting build and test information most helpful to .NET developers. Its goal is to make Pull Request results more actionable.
+The Build Analysis Check is a service to improve the GitHub Pull Request experience by highlighting build and test information most helpful to .NET developers. Its goal is to make Pull Request results more actionable.
```
Comment (markwilkie) [inline Documentation/Projects/Build Analysis/Introduction.md:3]: I like this - perhaps move the goal statement first?  (basically, just reverse the order)

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 
 ## What does it do?
```
[the remark this answers is not in the document]
Comment (missymessa) [reply Documentation/Projects/Build Analysis/Introduction.md:6]: @ulisesh what should we add in here about Known Issues?

File: Documentation/Projects/Build Analysis/Specifics.md [hunk 3]
```diff
@@ -12,4 +12,34 @@ The Build Analysis check includes these details:
   - Link to the Helix artifacts produced by a test
   - Link to the test execution details 
 
-Each Build Analysis page also includes a "was this helpful" link. This allows you to call out specific highlights or lowlights in your experience with the check. This feedback is then used by the Dev WF team to refine the analysis.
\ No newline at end of file
+Each Build Analysis page also includes a "was this helpful" link. This allows you to call out specific highlights or lowlights in your experience with the check. This feedback is then used by the Dev WF team to refine the analysis.
+
+## Known Issues
```
[the remark this answers is not in the document]
Comment (missymessa) [reply Documentation/Projects/Build Analysis/Specifics.md:17]: @ulisesh could you write a blurb to put in here about Known Issues. The users can click on the link to the Known Issue's doc for more information.

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 4]
```diff
@@ -14,14 +14,53 @@ For more details, see the [specifics](Specifics.md) document.
 
 ## How do I get it?
 
-Build Analysis is enabled on a per-repository basis. Contact dnceng to request it be enabled in your repository.
+Build Analysis is enabled on a per-repository basis. Contact [dnceng](mailto:dnceng@microsoft.com) to request it be enabled in your repository.
```
Comment (garath) [inline Documentation/Projects/Build Analysis/Introduction.md:17]:
```suggestion
Build Analysis is enabled on a per-repository basis. Contact [the engineering services team](https://github.com/dotnet/arcade/wiki/How-to-get-a-hold-of-.NET-Engineering-Services) to request it be enabled in your repository.
```

Consider linking to this wiki page that gives all contact options.

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 5]
```diff
@@ -14,14 +14,53 @@ For more details, see the [specifics](Specifics.md) document.
...
+
+**Note**: This feature requires your repositories pull request builds to occur in the `dnceng/public` Azure DevOps organization/project. Unfortunately, we cannot accommodate builds that occur in other Azure DevOps organizations (such as, DevDiv).
```
Comment (garath) [inline Documentation/Projects/Build Analysis/Introduction.md:19]:
```suggestion
**Note**: This feature requires your repository's pull request builds to occur in the `dnceng/public` Azure DevOps organization/project. Unfortunately, we cannot accommodate builds that occur in other Azure DevOps organizations (such as, DevDiv). 
```

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 6]
```diff
@@ -14,14 +14,53 @@ For more details, see the [specifics](Specifics.md) document.
...
+
+**Note**: This feature requires your repositories pull request builds to occur in the `dnceng/public` Azure DevOps organization/project. Unfortunately, we cannot accommodate builds that occur in other Azure DevOps organizations (such as, DevDiv). 
+
+### How do I know if it's enabled in my repository? 
+
+An easy way to see if Build Analysis is enabled in your repository is by viewing a pull request (particularly a recent one) and look for the `Build Analysis` check. 
+
+![build analysis check on pr](Resources/build_analysis_check_on_pr.png)
+
+Additionally, if you have sufficient permissions on your repository, you can browse to the Installed GitHub Apps settings and verify that the `.NET Helix` app is installed in your repository.
```
Comment (garath) [inline Documentation/Projects/Build Analysis/Introduction.md:27]: Just checking: This is done per-repository, and not at the GitHub organization level?

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 7]
```diff
@@ -14,14 +14,53 @@ For more details, see the [specifics](Specifics.md) document.
...
+
+As each pipeline completes, the Build Analysis Processor evaluates the results from the build (e.g. build failures, test failures) and updates the GitHub check run to reflect errors it's detected in your build. 
+
+Check out the [specifics](Specifics.md) page for a deeper dive into the features provided by Build Analysis. 
+
+## Other Features
+
+Related to the work done for Build Analysis, the following features were also built: 
+- [**Automatic Build Retries**](BuildRetryOnboard.md): Configure your repo's pull request builds to automatically retry when it encounters specific errors. 
+- [**Automatic Test Retries**](Test%20Retry%20Documentation.md): Configure tests in your repo's pull requests to automatically retry when they encounter specific errors. *This feature requires tests to use Helix as it leverages Helix to retry specific tests.*
```
Comment (garath) [inline Documentation/Projects/Build Analysis/Introduction.md:47]:
```suggestion
- [**Automatic Test Retries**](Test%20Retry%20Documentation.md): Configure helix-run tests in your repo's pull requests to automatically retry when they encounter specific errors. 
```

Reducing the emphasis on the Helix requirement. Still important, but not surprising or unreasonable.

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 8]
```diff
@@ -14,14 +14,53 @@ For more details, see the [specifics](Specifics.md) document.
...
 
 ## Frequently Asked Questions
 
 ### Is this useful even if my project does not use Helix?
 
 Yes! The Build Analysis result leverages Azure DevOps history and information to provide context and workflow improvements. Though it may bring additional information for Helix-based execution, it is not required. 
+
+### Why am I not seeing all the errors from my build/tests in Build Analysis? 
+
+Due to limitations, such as GitHub's character limits of the content of the check run, and throttling when making too many API calls to Azure DevOps for test history information, we are limited in how much data we can display on the check.
```
Comment (garath) [inline Documentation/Projects/Build Analysis/Introduction.md:58]:
```suggestion
Technical limitations, such as message size and API throttling, constrain the information shown in the check. The service tries to always include the most pertinent information with links to more complete views. 
```
Just a suggestion...

**Review by garath — APPROVED** [review]

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 9]
```diff
@@ -14,14 +14,53 @@ For more details, see the [specifics](Specifics.md) document.
...
+![installed github apps](Resources/installed_github_apps.png)
 
 ## How do I use it?
 
-Once enabled, a new GitHub check suite will be included in all pull requests. Navigate to the "Checks" tab, then look for the ".NET Helix" suite.
+Once enabled, a new GitHub check suite will be included in all pull requests. Navigate to the `Checks` tab, then look for the `.NET Helix` suite.
+
+![checks tab](Resources/checks_tab.png)
+
+The data that is used in the Build Analysis check is provided from each pipeline (in the dnceng/public Azure DevOps project) that your pull request runs through. From that data we are able to present you with errors that occurred during the build process, tests that failed, and even match the error messages with currently open [Known Issues](KnownIssues.md).
```
Comment (adiaaida) [inline Documentation/Projects/Build Analysis/Introduction.md:37]:
```suggestion
The data that is used in the Build Analysis check is provided by each pipeline (in the dnceng/public Azure DevOps project) that your pull request runs. From that data we are able to present you with errors that occurred during the build process, tests that failed, and even match the error messages with currently open [Known Issues](KnownIssues.md). 
```

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 10]
```diff
@@ -14,14 +14,53 @@ For more details, see the [specifics](Specifics.md) document.
...
 ## How do I use it?
 
-Once enabled, a new GitHub check suite will be included in all pull requests. Navigate to the "Checks" tab, then look for the ".NET Helix" suite.
+Once enabled, a new GitHub check suite will be included in all pull requests. Navigate to the `Checks` tab, then look for the `.NET Helix` suite.
+
+![checks tab](Resources/checks_tab.png)
+
+The data that is used in the Build Analysis check is provided from each pipeline (in the dnceng/public Azure DevOps project) that your pull request runs through. From that data we are able to present you with errors that occurred during the build process, tests that failed, and even match the error messages with currently open [Known Issues](KnownIssues.md). 
+
+As each pipeline completes, the Build Analysis Processor evaluates the results from the build (e.g. build failures, test failures) and updates the GitHub check run to reflect errors it's detected in your build.
```
Comment (adiaaida) [inline Documentation/Projects/Build Analysis/Introduction.md:39]:
```suggestion
As each pipeline completes, the Build Analysis Processor evaluates the results from the build (e.g. build failures, test failures) and updates the GitHub check run to reflect errors it detected in your build. 
```

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 11]
```diff
@@ -14,14 +14,53 @@ For more details, see the [specifics](Specifics.md) document.
...
+
+The data that is used in the Build Analysis check is provided from each pipeline (in the dnceng/public Azure DevOps project) that your pull request runs through. From that data we are able to present you with errors that occurred during the build process, tests that failed, and even match the error messages with currently open [Known Issues](KnownIssues.md). 
+
+As each pipeline completes, the Build Analysis Processor evaluates the results from the build (e.g. build failures, test failures) and updates the GitHub check run to reflect errors it's detected in your build. 
+
+Check out the [specifics](Specifics.md) page for a deeper dive into the features provided by Build Analysis. 
+
+## Other Features
+
+Related to the work done for Build Analysis, the following features were also built:
```
Comment (adiaaida) [inline Documentation/Projects/Build Analysis/Introduction.md:45]:
```suggestion
Related to the work done for Build Analysis, we also provide the following features: 
```

File: Documentation/Projects/Build Analysis/Introduction.md [hunk 12]
```diff
@@ -14,14 +14,53 @@ For more details, see the [specifics](Specifics.md) document.
...
+
+Due to limitations, such as GitHub's character limits of the content of the check run, and throttling when making too many API calls to Azure DevOps for test history information, we are limited in how much data we can display on the check. 
+
+However, we provide links from Build Analysis to logs and test results in Azure DevOps for deeper investigations. 
+
+### What benefit does Build Analysis provide over going directly to Azure DevOps?
+
+- One click to see errors that occurred on pipelines your pull request ran through. 
+- A single place to look at a combination of build and test errors. 
+- Be notified that an error that occurred on your pull request is related to a [Known Issue](KnownIssues.md).
```
Comment (adiaaida) [inline Documentation/Projects/Build Analysis/Introduction.md:66]:
```suggestion
- Notification that an error that occurred on your pull request is related to a [Known Issue](KnownIssues.md).
```

File: Documentation/Projects/Build Analysis/Specifics.md [hunk 13]
```diff
@@ -12,4 +12,34 @@ The Build Analysis check includes these details:
...
+
+Dive into the Known Issues documentation [here](KnownIssues.md).
+
+## Unique Build Failures
```
[the remark this answers is not in the document]
Comment (missymessa) [reply Documentation/Projects/Build Analysis/Specifics.md:21]: @AlitzelMendez @garath @ChadNedzlek anything you can think of that we should include in here for limitations on this? (I can link back to the blurb about technical limitations if that seems sufficient.)

**Review by adiaaida — APPROVED** [review]

**Comment by missymessa** [comment]
This PR has been open for quite some time and has information that is useful for customers. Merging this now. Outstanding feedback to be handled by: https://github.com/dotnet/arcade/issues/9814

**Pull request merged**